fe/fe-core/src/main/java/org/apache/doris/qe/cache/PartitionRange.java

```diff
@@ -238,7 +238,11 @@ public long realValue() {
         }
 
         private Date getDateValue(LiteralExpr expr) {
-            value = expr.getLongValue() / 1000000;
+            if (expr.getType().equals(Type.DATE)) {
+                value = expr.getLongValue();
+            } else {
+                value = expr.getLongValue() / DateLiteral.BASIC_ZERO_TIME;
```

Comment:
You already handle the type in `DateLiteral.getLongValue()`, why divide `BASIC_ZERO_TIME` here again?

Reply:
Yeah, you're right. Method `getDateValue` is only invoked in the case of `DATE`, the division is unnecessary.

Reply:
In `PartitionRange` class, the first parameter `type` of init-method cannot represent the second parameter `expr`'s real type, as illustrated in following figure:
![image](https://user-images.githubusercontent.com/20592822/144253131-dc2acc21-55f7-4966-b613-9b26a87b42e8.png)
 
So I reset to last commit, still handle the `expr`'s type with if-else statement, non-`DATE` type should divide `BASIC_ZERO_TIME`.

What about your opinion?